Classes/Domain/Model/Changes/Remove.php

```diff
@@ -104,4 +104,17 @@ public function apply(): void
             $this->feedbackCollection->add($updateParentNodeInfo);
         }
     }
+
+    private function getRemovalAttachmentPoint(): ?NodeAggregateId
```

Comment:
as pointed out here: https://github.com/neos/neos-development-collection/pull/4943#pullrequestreview-1949957107

this logic should imo reside in Neos.Neos so that there is an API way of deleting nodes so they will still be deletable via the workspace publisher.

Reply:
Indeed. Neos.Neos should provide a high level API for all change operations imho. But I would argue that this is out of scope for this PR.

We definitely need to straighten out the boundaries between Neos.Neos and Neos.Neos.Ui. I believe however that by now everything in Neos.Neos.Ui is treated as internal (mostly thanks to your effort :)), so it would be permissible for a deeper refactoring to take place post-release.

wdyt?